### ISSUE TITLE
ControlP5frame example doesn't work in Processing 3

### ISSUE THREAD

**Issue by ewelinka** [sojamo/controlp5#17]
I just installed Processing 3 and installed controlP5 library ( the newest version 2.2.5 ) . I tried to run the examples and I got error message by running examples -> extra -> ControlP5frame. The error message is : 
The function "add()" expects parameters like: "add(Component)" 
and the error line is the line 45:
  f.add(p);

With Processing 2.2.1 everything works fine.
I'm on Mac Yosemite, x64.

**Comment by ewelinka** [comment]
Why the issue was closed?

**Comment by sojamo** [comment]
sorry my bad. Processing 3 does not make use of java.awt anymore hence `add(Component)` will throw an error, see [things that may break your 2.x sketches](https://github.com/processing/processing/wiki/Changes-in-3.0) I will take a closer look at this and will update the frame examples accordingly.

**Comment by naabys** [comment]
Instead of adding the PApplet to a frame, I solved the issue by doing this in the constructor of my ControlFrame class (which still extends PApplet): 
```java
public ControlFrame(PApplet _parent, int _w, int _h, String _name) {
    super();
    PApplet.runSketch(new String[]{this.getClass().getName()}, this);
    parent = _parent;
    w=_w;
    h=_h;
}
```

It seems to work, so far, with my  controls using the  `plugTo(parent, "property")` method

**Comment by neilpanchal** [comment]
@naabys I tried your method and it crashes. If you can share your code, that'd be awesome. Thx!

**Comment by naabys** [comment]
Here's the class
```java
class ControlFrame extends PApplet {
  
  int w, h;
  PApplet parent;
  ControlP5 cp5;

  public ControlFrame(PApplet _parent, int _w, int _h, String _name) {
    super();   
    parent = _parent;
    w=_w;
    h=_h;
    PApplet.runSketch(new String[]{this.getClass().getName()}, this);
  }

  public void setup() {
    this.surface.setSize(w, h);
    cp5 = new ControlP5(this);
    cp5.addButton("Choose file").plugTo(parent,"chooseFile").setPosition(10,10);
    cp5.addButton("Save Frame").plugTo(parent, "saveFrame").setPosition(10,40);
    cp5.addToggle("auto").plugTo(parent, "auto").setPosition(10,70);
    cp5.addToggle("blend").plugTo(parent, "blend").setPosition(10,110);
    cp5.addNumberbox("seed").plugTo(parent, "seed").setRange(0, 360).setValue(1).setPosition(100, 10).setSize(100,20);
    cp5.addNumberbox("color").plugTo(parent, "c").setRange(0, 125).setValue(1).setPosition(100, 35).setSize(100,20);
    cp5.addNumberbox("color 2").plugTo(parent, "c2").setRange(0, 1000).setValue(1).setPosition(100, 60).setSize(100,20);
    
  }

}
```

and its instantiation in the `setup()` method of my sketch : 
```java
ControlFrame cf;
void setup(){
  cf = new ControlFrame(this, 400, 800, "Controls");
}
```

**Comment by sojamo** [comment]
@naabys thanks for this. just adding a sketch that works fine for me under processing 3 and controlP5 2.2.5. 

``` java
import controlP5.*;

ControlFrame cf;

float speed;
float pos;

void settings() {
  size(400,400,P3D);
}

void setup(){
  cf = new ControlFrame(this, 400, 800, "Controls");
  surface.setLocation(420,10);
  noStroke();
}

void draw() {
  pos += speed;
  background(0);
  translate(width/2,height/2);
  rotateY(pos);
  lights();
  box(100);
}




class ControlFrame extends PApplet {

  int w, h;
  PApplet parent;
  ControlP5 cp5;

  public ControlFrame(PApplet _parent, int _w, int _h, String _name) {
    super();   
    parent = _parent;
    w=_w;
    h=_h;
    PApplet.runSketch(new String[]{this.getClass().getName()}, this);
  }
  
  public void settings() {
    size(w,h);
  }
  
  public void setup() {
    surface.setLocation(10,10);
    cp5 = new ControlP5(this);
    cp5.addButton("Choose file").plugTo(parent,"chooseFile").setPosition(10,10);
    cp5.addButton("Save Frame").plugTo(parent, "saveFrame").setPosition(10,40);
    cp5.addToggle("auto").plugTo(parent, "auto").setPosition(10,70);
    cp5.addToggle("blend").plugTo(parent, "blend").setPosition(10,110);
    cp5.addNumberbox("seed").plugTo(parent, "seed").setRange(0, 360).setValue(1).setPosition(100, 10).setSize(100,20);
    cp5.addNumberbox("color").plugTo(parent, "c").setRange(0, 125).setValue(1).setPosition(100, 35).setSize(100,20);
    cp5.addNumberbox("color2").plugTo(parent, "c2").setRange(0, 1000).setValue(1).setPosition(100, 60).setSize(100,20);
    cp5.addSlider("speed").plugTo(parent, "speed").setRange(0, 0.1).setValue(0.01).setPosition(100, 260).setSize(200,20);
  }
  
  void draw() {
    background(40);
  }
}
```

**Comment by neilpanchal** [comment]
@sojamo @naabys Thanks :+1:

**Comment by naabys** [comment]
@neilpanchal once again, there are a lot of compatibility breaks from Processing2 to Processing3. ie: as @sojamo already told here `Processing 3 does not make use of java.awt anymore`. 
Besides, the code I shared earlier is more a workaround than the best way to do.
I am sorry not to have enough time to dig deeper in the lib (also, my Java is a bit rusty)

**Comment by sojamo** [comment]
fixed with version 2.2.6

**Comment by agrignard** [comment]
Maybe this link can be useful. I manage to make it work with Processing 3.3.7 and oscP5 2.2.6
https://forum.processing.org/two/discussion/13446/events-with-two-windows-using-controlp5-processing-3
